modules/ROOT/pages/dataweave-pattern-matching.adoc

```diff
@@ -1,27 +1,78 @@
-= Pattern Matching in DataWeave
+= Pattern Matching in DataWeave using a match statement
 ifndef::env-site,env-github[]
 include::_attributes.adoc[]
 endif::[]
 
-In DataWeave, pattern matching executes on the first pattern that matches the specified expression. DataWeave supports four different types of patterns:
+The *match* statement behaves like a match or switch statement in other languages, like Java or C++, and routes an input expression to a particular output expression based on some conditions. The syntax for a match statement is:
 
+.Match Statement
+[source,dataweave,linenums]
+----
+inputExpression match {
+  case <condition> -> <routing expression>
+  case <condition> -> <routing expression>
+  else -> <default routing expression>
+}
+----
+
+Like in other languages, the DataWeave match statement is an alternative and more compact way to organize multiple chained together if/else statments. A match statement consists of a list of case statements and perhaps a final else statement. Each case statement consists of a conditional selector expression that must evaluate to true or false, which corresponds to the expression you would write in an if statement's condition, followed by a correponding DataWeave routing expression that can include the match operator's input expression. 
+
+Each case statement can be an if statement, or can use other pattern matching shortcuts to define the case statement's condition. If a case statement is false, its routing expression is ignored. 
+
+DataWeave supports four different types of patterns for a case statement's condition:
+
+* <<pattern_match_expressions,Expressions>>
 * <<pattern_match_literals,Literal values>>
 * <<pattern_match_data_types,Data types>>
-* <<pattern_match_expressions,Expressions>>
 * <<pattern_match_regex,Regular expressions>>
 
-Each pattern type can be named or unnamed.
+Each case can be named or unnamed. If the case is named, the name stores the input expression as a local variable that can be used both in that case statement's conditional expression and in the corresponding routing expression. 
 
-.Pattern Matching Model
+.Match Statement Structure
 [source,dataweave,linenums]
 ----
 value match {
-  case (<name>:)?<pattern> -> <when matched>
-  case (<name>:)?<pattern> -> <when matched>
+  case (<name>:) <condition> -> <routing expression>
+  case (<name>:) <condition> -> <routing expression>
   else -> <when none of them matched>
 }
 ----
 
+Here are a few examples of case statements in a match statement. Other more detailed examples follow. 
+.Match Statement Example
+[source,dataweave,linenums]
+----
```

Comment:
need DW header.

Wondering if the example could be simplified or explained. It depends on people understanding precedence of the first matching case over the second (which would produce output if the first weren't there). People might also get hung up on trying to understand the 3rd case.